datafusion/common/src/config.rs

```diff
@@ -226,6 +226,11 @@ config_namespace! {
 
         /// Aggregate options
         pub aggregate: AggregateOptions, default = Default::default()
+
+        /// Fan-out during initial physical planning.
+        ///
+        /// This is mostly use to plan `UNION` children in parallel.
+        pub planning_concurrency: usize, default = 10
```

Comment:
It might be nice to default this to `num_cpus` like above?

Reply:
I think the config system currently doesn't allow dependent defaults (this would require a full DAG-style resolution).

Reply:
Github doesn't let me link, but online 216 is this:

```
        /// Number of partitions for query execution. Increasing partitions can increase
        /// concurrency. Defaults to the number of CPU cores on the system
        pub target_partitions: usize, default = num_cpus::get()
        ```